store/postgres/src/relational_queries.rs

```diff
@@ -1523,43 +1594,58 @@ impl QueryFragment<Pg> for FilterRange {
 /// The parallel to `EntityQuery`.
 ///
 /// Details of how query generation for `FilterQuery` works can be found
-/// in `docs/implementation/query-prefetching.md`
+/// `https://github.com/graphprotocol/rfcs/blob/master/engineering-plans/0001-graphql-query-prefetching.md`
 #[derive(Debug, Clone)]
 pub struct FilterQuery<'a> {
-    collection: FilterCollection<'a>,
-    sort_key: SortKey,
+    collection: &'a FilterCollection<'a>,
+    sort_key: SortKey<'a>,
     range: FilterRange,
     block: BlockNumber,
 }
 
 impl<'a> FilterQuery<'a> {
     pub fn new(
-        layout: &'a Layout,
-        collection: EntityCollection,
+        collection: &'a FilterCollection,
         filter: Option<&'a EntityFilter>,
         order: Option<(String, ValueType, EntityOrder)>,
         range: EntityRange,
         block: BlockNumber,
     ) -> Result<Self, QueryExecutionError> {
-        let collection = FilterCollection::new(layout, collection, filter)?;
-
         // Get the name of the column we order by; if there is more than one
         // table, we are querying an interface, and the order is on an attribute
         // in that interface so that all tables have a column for that. It is
         // therefore enough to just look at the first table to get the name
-        let first_table = collection
-            .first_table()
-            .expect("an entity query always contains at least one entity type/table");
+        let first_table = collection.first_table().unwrap();
```

Comment:
This is minor, but why remove the error message you get when we actually trip over this?